src/repositories/anchor-repository.ts

```diff
@@ -0,0 +1,39 @@
+import { EntityRepository, InsertResult } from "typeorm";
+import { BaseRepository } from "typeorm-transactional-cls-hooked";
+
+import Context from "../context";
+import Contextual from "../contextual";
+import { Anchor } from "../models/anchor";
+import { Request } from "../models/request";
+
+@EntityRepository(Anchor)
+export default class AnchorRepository extends BaseRepository<Anchor> implements Contextual {
```

Comment:
Can we add a comment describing what this class is for?  I believe these new repository classes are all about encapsulating the logic for talking to the postgres database that durably stores requests, right?

Reply:
It should be pretty explanatory what TypeORM repositories are. Here's the [link](https://typeorm.io/#/custom-repository) for more information.

I decoupled database handling logic from the services layer by using repositories. This is not related to the postgres database only since TypeORM provides the same interface for other providers as well.

Reply:
Maybe I just have a greater preference towards commenting than you do.  Personally I'm of the opinion that just about any class with actual logic in it should have a comment describing what it's responsible for.  If someone's new to this codebase and trying to figure out what this class does, they could see that AnchorRepository extends BaseRepository and implements Contextual, look up Contextual and eventually realize that just means its a singleton owned by the Context object, look up BaseRepository and see that it's imported from the 'typeorm' library, look up the typeorm library, find the documentation, and then realize that this is an ORM class for encapsulating database access logic, presumably for managing the persisted state related to "Anchors".  Or there could be a one sentence comment that just says that and saves them the effort.

I don't believe that someone who's trying to understand this codebase should be required to understand all the libraries that it is built on before they can even begin to start making sense of it.

Reply:
> > CAS Transactions is pretty vague as a commit message. Can you update it to something a bit more descriptive of what is actually changing here? Is there a github story that this can be linked to that explains the changes?
> 
> @stbrody have you read the PRs description, pls? I think it should be pretty clear what changes are there.

Oops, I missed that first comment.  My bad, sorry about that!